docs/examples/pp_cqm_presolve.rst

```diff
@@ -0,0 +1,93 @@
+.. _pp_cqm_presolve:
+
+===============================
+Preprocessing with CQM Presolve
+===============================
+
+This example uses explicit preprocessing to maximize performance when using
+a Leap classical-quantum hybrid Constrained Quadratic Model (CQM) solver.
+
+By default, Leap's CQM solver executes presolve algorithms in a preprocessing
+stage of its problem solving. You can locally run the presolve algorithms provided
+in Ocean prior to submitting your CQM to a solver, possibly reducing the size of
+the submitted problem and improving its structure.
+
+The purpose of this example is to show how to use Ocean's
+:class:`~dwave.preprocessing.presolve.Presolver` class.
+
+Example Requirements
+====================
+
+To run the code in this example, the following is required.
+
+* The requisite information for problem submission through SAPI, as described in :ref:`sapi_access`.
+* Ocean tools :doc:`dimod </docs_dimod/sdk_index>` and :doc:`dwave-preprocessing </docs_preprocessing/sdk_index>`.
+
+.. include:: hybrid_solver_service.rst
+  :start-after: example-requirements-start-marker
+  :end-before: example-requirements-end-marker
+
+Solution Steps
+==============
+
+.. include:: hybrid_solver_service.rst
+  :start-after: example-steps-start-marker
+  :end-before: example-steps-end-marker
+
+This example adds an optional step of preprocessing the submitted model.
+
+Formulate the Problem
+=====================
+
+This example uses a simplified problem for illustrative purposes: a CQM with no
+objective and just a single, one-variable constraint. This is sufficient to
+show how to run :class:`~dwave.preprocessing.presolve.Presolver` methods.
+
+The CQM created below has a constraint requiring that integer variable ``j``
+not exceed 5.
+
+.. testcode::
+
+    import dimod
+
+    j = dimod.Integer("j")
+    cqm = dimod.ConstrainedQuadraticModel()
+    cqm.add_constraint(j <= 5, "Maximum j")
```

Comment:
I would add an objective here, otherwise presolving is pretty pointless. How about:
```suggestion
    j = dimod.Integer("j")
    cqm = dimod.ConstrainedQuadraticModel()
    cqm.set_objective(-j)
    cqm.add_constraint(j <= 5, "Maximum j")
```

Reply:
Actually, an objective with `+j` works nicer for this example

Reply:
Why do you think? IMO, `-j` is nicer as it uses the visible (just defined) upper bound -- by converting it to offset.

Reply:
The example is to show the flow of actions so simpler is better, and with `+j` the lower bound is zero versus a high number, they don't have to do the switch in their head from minimizing a negative to maximizing a positive, and don't need to understand offset.